db/database.go

```diff
@@ -2323,3 +2103,206 @@ func (dbc *DatabaseContext) AuthenticatorOptions() auth.AuthenticatorOptions {
+				continue
+			}
+			provider.Name = name
+
+			// If this is the default provider, or there's only one provider defined, set IsDefault
+			if (db.Options.OIDCOptions.DefaultProvider != nil && name == *db.Options.OIDCOptions.DefaultProvider) || len(db.Options.OIDCOptions.Providers) == 1 {
+				provider.IsDefault = true
+			}
+
+			insecureSkipVerify := false
+			if db.Options.UnsupportedOptions != nil {
+				insecureSkipVerify = db.Options.UnsupportedOptions.OidcTlsSkipVerify
+			}
+			provider.InsecureSkipVerify = insecureSkipVerify
+
+			// If this isn't the default provider, add the provider to the callback URL (needed to identify provider to _oidc_callback)
+			if !provider.IsDefault && provider.CallbackURL != nil {
+				updatedCallback, err := auth.SetURLQueryParam(*provider.CallbackURL, auth.OIDCAuthProvider, name)
+				if err != nil {
+					return base.RedactErrorf("Failed to add provider %q to OIDC callback URL: %v", base.UD(name), err)
+				}
+				provider.CallbackURL = &updatedCallback
+			}
+
+			db.OIDCProviders[name] = provider
+		}
+	}
+
+	db.LocalJWTProviders = make(auth.LocalJWTProviderMap, len(db.Options.LocalJWTConfig))
+	for name, cfg := range db.Options.LocalJWTConfig {
+		db.LocalJWTProviders[name] = cfg.BuildProvider(name)
+	}
+
+	if db.UseXattrs() {
+		// Set the purge interval for tombstone compaction
+		db.PurgeInterval = DefaultPurgeInterval
+		cbStore, ok := base.AsCouchbaseBucketStore(db.Bucket)
+		if ok {
+			serverPurgeInterval, err := cbStore.MetadataPurgeInterval()
+			if err != nil {
+				base.WarnfCtx(ctx, "Unable to retrieve server's metadata purge interval - will use default value. %s", err)
+			} else if serverPurgeInterval > 0 {
+				db.PurgeInterval = serverPurgeInterval
+			}
+		}
+		base.InfofCtx(ctx, base.KeyAll, "Using metadata purge interval of %.2f days for tombstone compaction.", db.PurgeInterval.Hours()/24)
+
+		if db.Options.CompactInterval != 0 {
+			if db.autoImport {
+				db := Database{DatabaseContext: db}
+				// Wrap the dbContext's terminator in a SafeTerminator for the compaction task
+				bgtTerminator := base.NewSafeTerminator()
+				go func() {
+					<-db.terminator
+					bgtTerminator.Close()
+				}()
+				bgt, err := NewBackgroundTask(ctx, "Compact", func(ctx context.Context) error {
+					_, err := db.Compact(ctx, false, func(purgedDocCount *int) {}, bgtTerminator)
+					if err != nil {
+						base.WarnfCtx(ctx, "Error trying to compact tombstoned documents for %q with error: %v", db.Name, err)
+					}
+					return nil
+				}, time.Duration(db.Options.CompactInterval)*time.Second, db.terminator)
+				if err != nil {
+					return err
+				}
+				db.backgroundTasks = append(db.backgroundTasks, bgt)
+			} else {
+				base.WarnfCtx(ctx, "Automatic compaction can only be enabled on nodes running an Import process")
+			}
+		}
+
+	}
+
+	// Make sure there is no MaxTTL set on the bucket (SG #3314)
```

Comment:
I think we need to have this check db creation time, not online time, since we don't want expiries to be set on things like persistent config files.

Reply:
I guess both are required, to also catch the case where a bucket TTL is added after initial db creation but before startup.